Automattic/wp-calypso
# Jetpack connect plans first

This PR continues the work on https://github.com/Automattic/wp-calypso/pull/6843 . The rebase with current master have too many conflicts that it was easier to create a new branch from master and manually apply the changes. So I'm opening a new PR and closing that one.

This is the description of the original PR:

---

Something we want to try is to check how the results change if we show the jetpack plans selection screen to the users before the authorization, during the jetpack connect process, instead of doing it once the site is fully connected.

This PR implements this test.

![image](https://cloud.githubusercontent.com/assets/1554855/16928456/146f876e-4d33-11e6-86a9-50b1f61e947b.png)

![untitled screencast 18](https://cloud.githubusercontent.com/assets/1554855/16928683/3a42aa10-4d34-11e6-8056-57bba28b9e22.gif)
# How to test
1. Open http://calypso.localhost:3000/jetpack/connect
2. In your development console, write `localStorage.setItem('ABTests','{"jetpackConnectPlansFirst_20161024":"showPlansBeforeAuth"}')`
3. Connect one of your (disconnected) jetpack sites. After you enter your url, the next step you should see is the plans selection screen
4. Once you select a plan, the connection process should start
5. Once your site is connected, you should be redirected to the checkout with the plan you selected already in the cart
   5b. If you selected the free plan, you should be redirected to /posts instead

ping @rickybanister @oskosk @roccotripaldi @ryelle & @richardmuscat 

---

This PR now also adds three new routes:
http://calypso.localhost:3000/jetpack/connect/vaultpress
http://calypso.localhost:3000/jetpack/connect/akismet
http://calypso.localhost:3000/jetpack/connect/store

which would serve as a plans landing page for both logged and not logged user, and that allows you to pick a plan even before entering the url of your site

Test live: https://calypso.live/?branch=add/jetpack-plans-first
